### ISSUE TITLE
Reduce number of curl

### ISSUE THREAD

**Issue by tomdmr** [dr-ni/mi600#2]
Hi,
thanks for the basic idea, I changed the script to something like this:

#!/bin/bash
host=$1
auth=$2:$3
FILE=$(mktemp)
trap "rm -f $FILE" EXIT
curl -s -u "$auth" "$host"/status.html > $FILE
grep "webdata_now_p ="    $FILE | awk -F '"' '{print "Now:", $2, "W"}'
grep "webdata_today_e ="  $FILE | awk -F '"' '{print "Today:", $2, "kWh"}'
grep "webdata_total_e ="  $FILE | awk -F '"' '{print "Total:", $2, "kWh"}'

So you grab once the status to a temp-file, grep and awk it, and print everything at once. Or something like this:
pnow=$(grep "webdata_now_p ="    $FILE | awk -F '"' '{print "Now:", $2, "W"}')
etoday=$(grep "webdata_today_e ="  $FILE | awk -F '"' '{print "Today:", $2, "kWh"}')
# more stuff as needed...
now=$(date)
echo "$now\t$pnow\t$etoday"

**Comment by dr-ni** [comment]
Hi tomdr,

if you have a working improvement please make a pull request.

Regards, dr-ni

**Comment by int2018** [comment]
Hi,

here is a variant I made, the data access is in a loop, may that is not needed at all but anyway.

... initial stuff unchanged ...

ping -c3 $host > /dev/null 2>&1
if [ $? != 0 ]
then
  echo "Error: connection to $host failed"
exit 1
fi

for i in 1 2 3
do
  p=$(curl -s -u "$auth" "$host"/status.html | grep "$dat = " | awk -F '"' '{print $2}')
  if [ "$p" != "" ]
  then
    echo "$p $val"
    exit 0
  fi
  sleep 3
done

echo "Error: could not read $4"
exit 1
----------------------

But the main problem remains: The device does not update the value fast enough.
I have a simple energy meter attached to the ac outlet (Schuko) and it shows instant differences in the energy production but the result of the script (the inverter internal webbrowser) has still the same "old" value.
So, basically this approach to read the energy production is not working accurately due to the slow update of the inverter value,

Thanks anyway!

**Comment by dr-ni** [comment]
It's now included...
The slow update rate is caused by the inverters internal update rate.
So this cannot be improved here.

**Comment by dr-ni** [comment]
> But the main problem remains: The device does not update the value fast enough. I have a simple energy meter attached to the ac outlet (Schuko) and it shows instant differences in the energy production but the result of the script (the inverter internal webbrowser) has still the same "old" value. So, basically this approach to read the energy production is not working accurately due to the slow update of the inverter value,
> 
> Thanks anyway!

if you want more, this https://github.com/dr-ni/123solar_mi600/tree/main/misc/tools/deye-logger could be also an option

**Comment by tomdmr** [comment]
Yes, it is fairly slow. I thought it was due to bad connectivity, because it is quite a bit away from my WLAN-base. But a repeater close to it does not really help, so I conclude the logger is ...not really great.

**Comment by dr-ni** [comment]
the slow update rate is not a big problem because delayed sun and shadow period logging will be compensated over time.
Its a smoothed logging curve

**Comment by dr-ni** [comment]
https://github.com/dr-ni/123solar_mi600/tree/main/misc/tools/deye-logger

```+--------------------------------------+--------------------------------------+
| DCVoltagePV1          :     36.3 V   | DCVoltagePV2         :      34.1 V   |
| DCCurrentPV1          :      0.4 A   | DCCurrentPV2         :       6.8 A   |
| DCPowerPV1            :     14.5 W   | DCPowerPV2           :     231.9 W   |
| DailyProductionPV1    :      0.5 kWh | DailyProductionPV2   :       0.8 kWh |
| TotalProductionPV1    :    309.0 kWh | TotalProductionPV2   :     230.6 kWh |
+--------------------------------------+--------------------------------------+
| DCVoltagePV3          :      0.0 V   | DCVoltagePV4         :       0.0 V   |
| DCCurrentPV3          :      0.0 A   | DCCurrentPV4         :       0.0 A   |
| DCPowerPV3            :      0.0 W   | DCPowerPV4           :       0.0 W   |
| DailyProductionPV3    :      0.0 kWh | DailyProductionPV4   :       0.0 kWh |
| TotalProductionPV3    :      0.0 kWh | TotalProductionPV4   :       0.0 kWh |
+--------------------------------------+--------------------------------------+
| CurrentPower          :    240.0 W   | InverterSerialNumber :    xxxxxxxx |
| YieldToday            :      1.4 kWh | InverterRatedPower   :     600.0 W   |
| TotalYield            :    562.8 kWh | NumberOfMPPTs        :         2     |
| ACVoltage1            :    240.0 V   | NumberOfPhases       :         1     |
| ACCurrent1            :      0.0 A   | LoggerSerialNumber   :    xxxxxxxxxxx |
| ACOutputFrequency     :     50.0 Hz  | LoggerMacAddress     :  xxxxxxxxxxxx |
| ACRadiatorTemperature :     19.1 GrdC| DataTimeStamp  : 2023-02-11 15:13:03 |
+--------------------------------------+--------------------------------------+
```
